### ISSUE TITLE
Typescript build fails when using isolatedModules flag

### ISSUE THREAD

**Issue by jrbarnard** [optimizely/javascript-sdk#609]
##  What I wanted to do

I tried to build a project using this lib with typescript. I have the isolatedModules flag set to true due to the use of babel.

## What I expected to happen

It to build correctly

## What actually happened

I got the error:
```
../node_modules/@optimizely/optimizely-sdk/lib/core/event_processor/index.ts:25:10
Type error: Re-exporting a type when the '--isolatedModules' flag is provided requires using 'export type'.

  23 | }
  24 | 
> 25 | export { EventProcessor, LocalStoragePendingEventsDispatcher } from '@optimizely/js-sdk-event-processor';
     |          ^
  26 | 
  27 | export default { createEventProcessor, LocalStoragePendingEventsDispatcher };
```

## Steps to reproduce
SDK version: 3.2.2
Node version: v12.18.0
NPM version: 6.14.4
Typescript version: 3.9.7

Use tsconfig with isolatedModules set to true
Run tsc
See error

**Comment by jrbarnard** [comment]
A quick update on this, I've bypassed the issue by explicitly stopping typescript picking up .ts files within node_modules but I believe the issue may still stand

**Comment by yavorona** [comment]
Thanks @jrbarnard for reporting this issue. We merged a [fix](https://github.com/optimizely/javascript-sdk/pull/610) and released it in [v4.4.1](https://github.com/optimizely/javascript-sdk/releases/tag/v4.4.1)
